RecoBTag/ONNXRuntime/plugins/DeepBoostedJetTagsONNXProducer.cc

```diff
@@ -0,0 +1,298 @@
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/Framework/interface/makeRefToBaseProdFrom.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/BTauReco/interface/JetTag.h"
+
+#include "DataFormats/BTauReco/interface/DeepBoostedJetTagInfo.h"
+
+#include "PhysicsTools/ONNXRuntime/interface/ONNXRuntime.h"
+
+#include <iostream>
+#include <fstream>
+#include <algorithm>
+#include <numeric>
+
+using namespace Ort;
+
+// Hold the ONNX runtime object in the edm::GlobalCache.
+struct ONNXRuntimeCache {
+  ONNXRuntimeCache() : onnx_runtime(nullptr) {}
+
+  std::atomic<ONNXRuntime *> onnx_runtime;
+};
+
+// struct to hold preprocessing parameters
+struct PreprocessParams {
+  struct VarInfo {
+    VarInfo() {}
+    VarInfo(float median, float norm_factor) : center(median), norm_factor(norm_factor) {}
+    float center = 0;
+    float norm_factor = 1;
+  };
+
+  unsigned var_length = 0;
+  std::vector<std::string> var_names;
+  std::unordered_map<std::string, VarInfo> var_info_map;
+
+  VarInfo get_info(const std::string &name) const {
+    auto item = var_info_map.find(name);
+    if (item != var_info_map.end()) {
+      return item->second;
+    } else {
+      throw cms::Exception("InvalidArgument") << "Cannot find variable info for " << name;
+    }
+  }
+};
+
+class DeepBoostedJetTagsONNXProducer : public edm::stream::EDProducer<edm::GlobalCache<ONNXRuntimeCache>> {
```

Comment:
Actually why not just using the `ONNXRuntime` directly in the `edm::GlobalCache`?

Reply:
Good point! 
Actually this part was copied from the original TF-based DeepJet tagger and I was not sure why it was implemented in that way...